tests/ha/ha_cluster_preflight_check.pm

```diff
@@ -0,0 +1,137 @@
+    my $cluster_name = get_cluster_name;
+
+    # Install needed package
+    $self->select_serial_terminal;
+    zypper_call 'in python3-cluster-preflight-check';
+
+    # Ensure that the cluster state is correct before executing the checks
+    check_cluster_state;
+
+    # We have to wait for previous nodes to finish the tests, as they can't be done in parallel without any damages!
+    barrier_wait("PREFLIGHT_CHECK_INIT_${cluster_name}_NODE$_") foreach (1 .. (get_node_index) - 1);
+
+    # List of things to check
+    my @checks    = qw(env-check cluster-check kill-sbd kill-corosync kill-pacemaker split-brain-iptables);
+    my @can_fence = qw(kill-sbd kill-corosync kill-pacemaker split-brain-iptables);
+
+    # Loop on each check
+    my $preflight_start_time = time;
+    foreach my $check (@checks) {
+        # Is the check can trigger a fencing?
+        my $trigger_fencing = any { $_ eq $check } @can_fence;
+
+        # Execute the command
+        my $cmd = "ha-cluster-preflight-check --yes --${check}";
+        record_info("${check}", "Executing ${cmd}");
+        my $cmd_fails = script_run "${cmd}";
+        record_info('ERROR', "Failure while executing '$cmd'", result => 'fail') unless ((defined $cmd_fails and $cmd_fails == 0) or $trigger_fencing);
+        save_screenshot;
+
+        # Some commands may lead to a reboot of the node
+        if ($trigger_fencing) {
+            my $loop_count = bmwqemu::scale_timeout(15);    # Wait 1 minute (15*4) maximum, can be scaled with SCALE_TIMEOUT
+            while (1) {
+                last if ($loop_count-- <= 0);
+                if (check_screen('bootloader-grub2', 0, no_wait => 1)) {
+                    # Wait for boot and reconnect to root console
+                    $self->wait_boot;
+                    $self->select_serial_terminal;
+                    last;
+                }
+                sleep 4;
+            }
+        }
+    }
+    my $preflight_end_time = time;
+    upload_preflight_check_logs;
+
+    # Parse the logs to get a better overview in openQA
+    my $results_file = '/tmp/preflight_cluster.json';
+    my %results;
+    $results{tests}                = [];
+    $results{info}                 = {};
+    $results{summary}              = {};
+    $results{info}->{timestamp}    = time;
+    $results{info}->{distro}       = "";
+    $results{info}->{results_file} = "";
+    $results{summary}->{num_tests} = 0;
+    $results{summary}->{passed}    = 0;
+    $results{summary}->{duration}  = $preflight_end_time - $preflight_start_time;
+
+    my $dir_log = '/var/lib/ha-cluster-preflight-check/';
+    my $output  = script_output("for FILE in $dir_log/*.report; do awk -F':' '/Testcase:|ERROR:/ { print }' ORS=' ' \$FILE 2>/dev/null && echo; done");
+    foreach my $line (split("\n", $output)) {
+        my @tab = split(/\s+/, $line);
+        my %aux = ();
+        $results{summary}->{num_tests}++;
+        $aux{name}       = 'force_kill_' . lc($tab[3]);
```

Comment:
What about `$aux{name} = lc(join('_', $tab[1], $tab[2], $tab[3]));` ?

That way code will work with other types of tests that are not *Force kill something* without needing to change that line of code.

Reply:
Good point!